techmatters/flex-plugins
# CSAM reporting cleanup

Primary reviewer: @stephenhand 
Note: easier to review commit by commit.

## Description
This PR is a UI/UX cleanup of the CSAM report components.
Please read the below linked Jira card for the full detail on this changes. Other than those, the only really relevant change is the inclusion of a new input type radio.

Special considerations:
- One of the items of the ticket is "The italic font on the summary screen should pull the real italic font, not the oblique version of it". After investigating, it looks like we will need to import a new font family just to address this, and after a small talk with Dee she's ok leaving it as they are.
- The regex validation for url is not the most exact one, but I couldn't find a valid url that's not accepted. The inverse might be true tho (not restrictive enough). Happy to get some help if someone has more expertise in rexeg.
- I'm hiding the overall label for the radio button set to match the designs. This is not very accessible tho. Happy to change if the design is reviewed.

### Checklist
- [x] Corresponding issue has been opened https://bugs.benetech.org/browse/CHI-977

### Verification steps
- Run Flex locally (`npm run dev`).
- Open a new CSAM report (top right button of the contact form).
- Confirm that behavior is the expected, validations are as described in the ticket and the functionality to submit reports is still working.
